hachoir/editor/field.py

```diff
@@ -63,7 +63,7 @@ def writeInto(self, output):
         addr = self._parent._getFieldInputAddress(self._name)
         input = self._parent.input
         stream = input.stream
-        if size % 8:
+        if size % 8 or addr % 8:
             output.copyBitsFrom(stream, addr, size, input.endian)
         else:
             output.copyBytesFrom(stream, addr, size // 8)
```

Comment:
Maybe copyBitsFrom() can be always called, if copyBitsFrom() is modified to accept addressed not aligned on bytes, no?

Reply:
I think that we would still be solving the same problems then, but one level deeper. That would clean up the FakeField class.

Reply:
At the very bottom of the chain, we still have the problem that byte transfer is usually faster than bit transfer, and that we may have unaligned streams (however, I think bitstreams usually would try to align themselves now and then for these same reasons).